### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by parberge**
* Removed faulty test
* Refactor existing

We want to make sure we test things more isolated (units).
Fake the data isolated instead of using `test_data.py`.

Other small fixes.

### CODE REVIEW

File: tests/test_timereport.py [hunk 1]
```diff
@@ -1,76 +1,29 @@
...
+    assert isinstance(slack_payload_extractor({'body': 'fake=fake'}), dict)
 
-    for i in db.scan((db.user_name == 'kamger') & (db.event_date.between(event_date_start, event_date_end))):
-        result.append(i.attribute_values)
 
-    assert len(result) == 3
+def test_factory():
+    fake_order = dict(user_id='fake', user_name='fake mcFake', text='fake_cmd=do_fake+fake_reason+2018-01-01')
+    fake_result = factory(fake_order)
+    assert isinstance(fake_result, list)
```
Comment (kamigerami) [inline tests/test_timereport.py]: so why do we only test if they are instance of something, strings, lists dicts or datetime objects.. why don't we test that they actually are correct values provided .. for example .. if reason is one of ['vab,'ledigt','sjuk'] etc ..  or action is one of add/list/show etcc?

Comment (parberge) [reply tests/test_timereport.py]: Because this test is about the factory function. Verification of actions etc takes place somewhere else.

Comment (parberge) [reply tests/test_timereport.py]: And there is no real value to test that the value foo matches foo. We only want to make sure it's the expected type

Comment (parberge) [reply tests/test_timereport.py]: @kamigerami Are you satisfied with my explanation? If so please mark "resolve conversation", or if you have a suggestion on what you would like me to change, please let me know.

**Comment by parberge** [comment]
@kamigerami @tommymalmqvist What say you? 😸

**Pull request merged**